### ISSUE TITLE
Entity nested with other entity collection

### ISSUE THREAD

**Issue by danilobicas** [fabriciocolombo/delphi-rest-client-api#109]
Hi.
follow the entities
class moviment
            mov_codigo:int;
            collection<itens> itens;
class itens
    ite_codigo: int;
   produto: produtcs;

class products
pro_codigo:int;
pro_value:decimal;

When I receive my json from webapi Delphi show me a message: Class T not found. when I click OK it show next message(more specific): UnMarshalling error for field "moviment.itens" : class T not found.

In my class in delphi I do as follow:

class moviment
            mov_codigo:int;
             itens: TObjectList;
class itens
    ite_codigo: int;
   produto: TObject;;

class products
pro_codigo:int;
pro_value:decimal;
I thinks this is the error but I don't know how I fix this.

Please help!!!!

**Comment by thomaserlang** [comment]
Your code is rather hard to read.  
Try something like this:

```
itens: TObjectList<itens>;
```
You need to tell delphi what type of object the list will contain.

**Comment by danilobicas** [comment]
Yes but my delphi is 2007 and don't accept TObjectList<Itens>....
Unfortunately

Tks

Danilo

2018-02-22 19:37 GMT-03:00 Thomas Erlang <notifications@github.com>:

**Comment by danilobicas** [comment]
follow below my delphi code
`MovimentoProdutoViewModel = class
 private
    Fmov_codigo: integer;
    Fmop_quantidade: currency;
    Fpro_codigo: integer;
    Flab_produtos: TObject;
 protected
 public
    constructor Create;
    destructor Destroy;
 published
    property pro_codigo:integer read Fpro_codigo write Fpro_codigo;
    property mov_codigo:integer read Fmov_codigo write Fmov_codigo;
    property mop_quantidade:currency read Fmop_quantidade write Fmop_quantidade;
    property ProdutosViewModel:TObject read Flab_produtos write Flab_produtos;
end;

MovimentoViewModel = class

   private
    Fmov_codigo: integer;
    Fmov_data: string;
    Ftpm_codigo: integer;
    Fmov_descricao: string;
    Fmov_nota: string;
    Flab_movimentoprodutoviewmodel: TObjectList;
    FTipoMovimentoViewModel: TObject;

   protected

   public
     constructor Create;
     destructor Destroy;
   published
     property mov_codigo: integer read Fmov_codigo write Fmov_codigo;
     property tpm_codigo: integer read Ftpm_codigo write Ftpm_codigo;
     property mov_data:string read Fmov_data write Fmov_data;
     property mov_nota:string read Fmov_nota write Fmov_nota;
     property mov_descricao:string read Fmov_descricao write Fmov_descricao;
     property MovimentoProdutoViewModel: TObjectList read Flab_movimentoprodutoviewmodel write Flab_movimentoprodutoviewmodel;
     property TipoMovimentoViewModel: TObject read FTipoMovimentoViewModel write FTipoMovimentoViewModel;
end;

constructor MovimentoViewModel.Create;
begin
  Flab_movimentoprodutoviewmodel:= TObjectList.Create;
  FTipoMovimentoViewModel:=TObjectList.Create;
end;

destructor MovimentoViewModel.Destroy;
begin
  Flab_movimentoprodutoviewmodel.Destroy;
  FTipoMovimentoViewModel.Destroy;
end;
`
Previously I said that my delphi is 2007 and I can´t put TObjectList<MovimentoProdutoViewModel>

can you help me?
